pkg/metrics/logging_hook.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/cilium/cilium/pkg/components"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/sirupsen/logrus"
+)
+
+// incErrorWarningMetric increments the error/warning Prometheus metric based on
+// the logger entry.
+func incErrorWarningMetric(entry *logrus.Entry) error {
+	// Every instance of Entry created from DefaultLogger is expected to
+	// have 'subsystem' field.
+	iSubsystem, ok := entry.Data[logfields.LogSubsys]
+	if !ok {
+		serializedEntry, err := entry.String()
```

Comment:
If we are doing a pre-check (following commit) then this can probably be a `panic`. (I don't like panics so I leave for others to give opinion about it)

Reply:
@joestringer @tgraf - your thoughts about `panic`? I'm OK with it.